src/Diagnostics.RuntimeHost/Controllers/DiagnosticControllerBase.cs

```diff
@@ -808,9 +808,14 @@ private bool VerifyEntity(EntityInvoker invoker, ref QueryResponse<DiagnosticApi
         }
 
         private Definition RemovePIIFromDefinition(Definition definition, bool isInternal)
-        {
-            if (!isInternal) definition.Author = string.Empty;
-            return definition;
+        { 
+            string definitionString = JsonConvert.SerializeObject(definition);
+            Definition definitionCopy = JsonConvert.DeserializeObject<Definition>(definitionString);
+            if (!isInternal)
+            {
+                definitionCopy.Author = string.Empty;
```

Comment:
May need to null check `definition`.

Also, this raises the opportunity for a `CloneJson` extension method like the one described here: https://stackoverflow.com/questions/78536/deep-cloning-objects

Reply:
We will want to revisit the clone logic for the diagrole for the root fix. Thanks Daniel!